prefix/match.go

```diff
@@ -0,0 +1,32 @@
+package prefix
+
+import (
+	"path/filepath"
+	"strings"
+)
+
+func Match(pattern, name string) (bool, bool) {
+	count := strings.Count(name, string(filepath.Separator))
```

Comment:
I think you are right about this. I guess https://github.com/moby/buildkit/pull/2082/files#diff-364fe775b55b3eec6b3ea0859ab920f0981c61d377e47830a46de5ec04058c44R437 does not fail on Windows because there are no test cases that depend on parent dirs being included in the checksum. Not sure if there's a way to simulate a change in parent dir metadata in that test so we can capture that scenario.